lib/endpoints/class-wp-rest-comments-controller.php

```diff
@@ -720,7 +823,7 @@ protected function prepare_item_for_database( $request ) {
 		}
 
 		if ( isset( $request['parent'] ) ) {
-			$prepared_comment['comment_parent'] = $request['parent'];
+			$prepared_comment['comment_parent'] = (int) $request['parent'];
```

Comment:
I've added `(int)` because somehow `$request['parent'']` was an empty array in some tests (e.g. `test_get_items_no_post()`. I haven't searched yet, why (it is not indicated by the test itself though) and I am not sure if there is a bug behind, which should be fixed. I will have a look, but maybe someone knows already why it is sometimes an array.